src/NuGet.Clients/NuGet.PackageManagement.UI/Models/Package/PackageModel.cs

```diff
@@ -9,7 +10,22 @@ namespace NuGet.PackageManagement.UI
 {
     public abstract class PackageModel
     {
-        public PackageIdentity Identity { get; internal set; }
+        protected PackageModel(PackageIdentity identity, string title, string description, string authors, Uri projectUrl, string[] tags, DateTimeOffset? published)
+        {
+            if (identity == null)
+            {
+                throw new ArgumentNullException(nameof(identity));
+            }
+            Identity = identity;
+            Title = title;
+            Description = description;
+            Authors = authors;
+            ProjectUrl = projectUrl;
+            Tags = tags;
+            Published = published;
+        }
+
+        public PackageIdentity Identity { get; private set; }
```

Comment:
If set in the constructor, we don't need to have a private setter. This implies the internal implementation can change it after it's constructed. We should be explicit about the fields we allow to be changed versus the ones that are readonly.